src/Core/Services/MetadataProviders/IMetadataProviderFactory.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Azure.DataApiBuilder.Core.Services.MetadataProviders
+{
+    /// <summary>
+    /// IMetadataProviderFactory class.
+    /// Used to get the appropriate metadata provider based on the data source name.
+    /// </summary>
+    public interface IMetadataProviderFactory
+    {
+        /// <summary>
+        /// Gets the appropriate metadata provider based on the data source name.
+        /// </summary>
+        /// <param name="dataSourceName">dataSourceName.</param>
+        /// <returns>ISqlMetadataProvider.</returns>
```

Comment:
The param name/returns is not really helpful here. Can either populate properly or remove.